File: menu.c
#include <stdio.h>
#include "menu.h"
#include "tela.h"
#include "informacoes.h"

void mostrarMenu(void) {

    int opcao;

    do{

        limparTela();

        printf("\n");
        printf("====================================================\n");
        printf("|                                                  |\n");
        printf("|                    VESTIGIUM                     |\n");
        printf("|                                                  |\n");
        printf("|                  1. INICIAR                      |\n");
        printf("|                  2. INFORMAÇÕES                  |\n");
        printf("|                  3. CREDITOS                     |\n");
        printf("|                  4. SAIR                         |\n");
        printf("|                                                  |\n");
        printf("====================================================\n");

        printf("\nEscolha uma opção: ");
        scanf("%d", &opcao);
        
        while (getchar() != '\n');
        
        switch (opcao) {
            case 1:
                printf("\nIniciando jogo...\n");
                esperarEnter();
                break;

            case 2:
                mostrarInformacoes();
                break;

            case 3:
                printf("\nCreditos\n");
                esperarEnter();
                break;

            case 4:
                printf("\nSaindo do jogo...\n");
                break;

            default:
                printf("\nOpção invalida!\n");
        }
       
    } while (opcao != 4);
}

File: Menu.java
public class Menu {
    public static void mostrarMenu() {
        int opcao;
        do {
            Tela.limparTela();
            System.out.println("╔══════════════════════════════════════╗");
            System.out.println("║              VESTIGIUM               ║");
            System.out.println("╠══════════════════════════════════════╣");
            System.out.println("║  1. Iniciar Jogo                     ║");
            System.out.println("║  2. Informações                      ║");
            System.out.println("║  3. Créditos                         ║");
            System.out.println("║  4. Sair                             ║");
            System.out.println("╚══════════════════════════════════════╝");
            System.out.print("\nEscolha uma opção: ");
            opcao = Tela.lerInteiro();
            switch (opcao) {
                case 1 -> Jogo.iniciarJogo();
                case 2 -> Informacoes.mostrarInformacoes();
                case 3 -> Creditos.mostrarCreditos();
                case 4 -> { Tela.limparTela(); System.out.println("Encerrando VESTIGIUM..."); }
                default -> { System.out.println("\nOpção inválida."); Tela.esperarEnter(); }
            }
        } while (opcao != 4);
    }
}
